### PR TITLE
fix: show error message when measureText API breaks in brave

### LINKED ISSUE #6325: Revisited my previous drawing and all the text is now broken.
This is the non-logged in version. Just stored in session state. I tried messing with the settings and can't get the text to go back to normal. It's almost like it's masked. But it's also disconnected from the shape now. 

image

### PR DESCRIPTION
**Pull request by ad1992**
fixes #6325 
- [x] Update error comp to use children
- [x] Use i18n
- [x] Add docs for turning off aggressive filtering
- [x] tweak error message
- [x] Add specs

In brave browser when "Aggressive Block fingerprinting" is turned on, the `measureText` API breaks and all text elements break. Unless you turn off the filtering, there isn't any way to fix the elements in Brave browser

https://user-images.githubusercontent.com/11256141/223751831-7206559e-7638-44d0-bd57-8d09b755e749.mp4

Checking the width computes to 0

image

This PR shows an error dialog when it detects brave browser and the measureText API not working

Screenshot 2023-03-08 at 10 34 47 PM

### CODE REVIEW

File: dev-docs/docs/@excalidraw/excalidraw/faq.mdx [hunk 1]
```diff
@@ -4,6 +4,31 @@
 
 No, Excalidraw package doesn't come with collaboration built in, since the implementation is specific to each host app. We expose APIs which you can use to communicate with Excalidraw which you can use to implement it. You can check our own implementation [here](https://github.com/excalidraw/excalidraw/blob/master/src/excalidraw-app/index.tsx). Here is a [detailed answer](https://github.com/excalidraw/excalidraw/discussions/3879#discussioncomment-1110524) on how you can achieve the same.
 
+### Turning off Aggresive Block Fingerprinting in Brave browser
```
Comment (dwelle) [inline dev-docs/docs/@excalidraw/excalidraw/faq.mdx:7]:
```suggestion
### Turning off Aggressive Anti-Fingerprinting in Brave browser
```

File: dev-docs/docs/@excalidraw/excalidraw/faq.mdx [hunk 2]
```diff
@@ -4,6 +4,31 @@
...
+
+Since when *Aggresive Block Fingerprinting* is turned on, the `measureText` API breaks which in turn breaks the Test Elements in your drawings. Here is more [info](https://github.com/excalidraw/excalidraw/pull/6336) on the same.
```
Comment (dwelle) [inline dev-docs/docs/@excalidraw/excalidraw/faq.mdx:9]:
```suggestion
When *Aggressive Anti-Fingerprinting* is turned on, the `measureText` API breaks which in turn breaks the Text Elements in your drawings. Here is more [info](https://github.com/excalidraw/excalidraw/pull/6336) on the same.
```

File: dev-docs/docs/@excalidraw/excalidraw/faq.mdx [hunk 3]
```diff
@@ -4,6 +4,31 @@
...
+
+Hence we have added a *warning* when such situtation is encountered.
```
Comment (dwelle) [inline dev-docs/docs/@excalidraw/excalidraw/faq.mdx:11]:
```suggestion
```

redundant

File: dev-docs/docs/@excalidraw/excalidraw/faq.mdx [hunk 4]
```diff
@@ -4,6 +4,31 @@
...
+
+We strongly recomment turning it off. You can follow :point_down: steps on how to do so.
```
Comment (dwelle) [inline dev-docs/docs/@excalidraw/excalidraw/faq.mdx:13]:
```suggestion
We strongly recommend turning it off. You can follow the steps below on how to do so.
```

File: dev-docs/docs/@excalidraw/excalidraw/faq.mdx [hunk 5]
```diff
@@ -4,6 +4,31 @@
...
+
+1. Open [excalidraw.com](https://excalidraw.com) in Brave and click on the `Sheild` button
```
Comment (dwelle) [inline dev-docs/docs/@excalidraw/excalidraw/faq.mdx:15]:
```suggestion
1. Open [excalidraw.com](https://excalidraw.com) in Brave and click on the **Shield** button
```

File: dev-docs/docs/@excalidraw/excalidraw/faq.mdx [hunk 6]
```diff
@@ -4,6 +4,31 @@
...
+
+![Aggresive block fingerprinting](../../assets/aggressive-block-fingerprint.png)
+
+3. Switch to **Block Fingerprinting**
+
+![Block filtering](../../assets/block-fingerprint.png)
+
+4. Thats all. All text elements should be fixed now 🎉
+
+If disabling this setting doesn't fix the display of text elements, please consider opening an [issue](https://github.com/excalidraw/excalidraw/issues/new) on our GitHub or join write to us on [Discord](https://discord.gg/UexuTaE).
```
Comment (dwelle) [inline dev-docs/docs/@excalidraw/excalidraw/faq.mdx:28]:
```suggestion
If disabling this setting doesn't fix the display of text elements, please consider opening an [issue](https://github.com/excalidraw/excalidraw/issues/new) on our GitHub, or message us on [Discord](https://discord.gg/UexuTaE).
```

File: src/constants.ts [hunk 7]
```diff
@@ -12,6 +12,11 @@ export const isFirefox =
 export const isChrome = navigator.userAgent.indexOf("Chrome") !== -1;
 export const isSafari =
   !isChrome && navigator.userAgent.indexOf("Safari") !== -1;
+// keeping function so it can be mocked in test
+export const isBrave = () =>
+  (navigator as any).brave &&
+  (navigator as any).brave.isBrave &&
+  (navigator as any).brave.isBrave.name === "isBrave";
```
Comment (dwelle) [inline src/constants.ts:19]:
```suggestion
export const isBrave = () =>
  (navigator as any).brave?.isBrave?.name === "isBrave";
```

File: src/locales/en.json [hunk 8]
```diff
@@ -206,7 +205,22 @@
...
+      "start": "Looks like you are using Brave browser with",
+      "aggressive_block_fingerprint": "Aggressively Block Fingerprinting",
+      "setting_enabled": "setting enabled",
+      "break": "This could result in breaking the",
+      "text_elements": "Text Elements",
+      "in_your_drawings": "in your drawings",
+      "strongly_recommend": "We strongly recommend disabling this setting. You can follow",
+      "steps": "these steps",
+      "how": "on how to do so",
+      "disable_setting": " If disabling this setting doesn't fix the display of text elements, please open",
```
Comment (dwelle) [inline src/locales/en.json:219]:
```suggestion
      "disable_setting": " If disabling this setting doesn't fix the display of text elements, please open an",
```

Comment (dwelle) [inline src/locales/en.json:219]:
```suggestion
      "disable_setting": "If disabling this setting doesn't fix the display of text elements, please open",
```

File: dev-docs/docs/@excalidraw/excalidraw/faq.mdx [hunk 9]
```diff
@@ -4,6 +4,31 @@
...
+Hence we have added a *warning* when such situtation is encountered.
+
+We strongly recomment turning it off. You can follow :point_down: steps on how to do so.
+
+1. Open [excalidraw.com](https://excalidraw.com) in Brave and click on the `Sheild` button
+![sheild Button](../../assets/brave-sheild.png)
+
+2. Once opened, look for **Aggressively Block Fingerprinting**
+
+![Aggresive block fingerprinting](../../assets/aggressive-block-fingerprint.png)
```
[the remark this answers is not in the document]
Comment (ad1992) [reply dev-docs/docs/@excalidraw/excalidraw/faq.mdx:21]: yep done, let me know if its good now

**Review by DanielJGeiger — COMMENTED** [review]
I tested this PR in Brave.  I got the error message with aggressive fingerprint blocking but did not get it with standard fingerprint blocking.

**Review by dwelle — APPROVED** [review]

**Pull request merged**